### PR TITLE
[ExoBundle] Add score on steps

### LINKED ISSUE #534: [ExoBundle] Score-Etape vs Résultats
Le score de l'étape devrait indiquer le résultats total acquis depuis le début de l'exercice

et ainsi être identique/en cohérence avec le score des résultats

### PR DESCRIPTION
**Pull request by pitrackster**
Show step score in exercise player and paper view
resolve #534

### CODE REVIEW

File: plugin/exo/Resources/modules/paper/Partials/show.html [hunk 1]
```diff
@@ -21,13 +21,27 @@ <h4 class="paper-title">
...
+    </div>
+<!--
     <question-show data-ng-repeat="question in paperShowCtrl.questions"
                    data-question="question"
                    data-question-paper="paperShowCtrl.getQuestionPaper(question)"
                    data-include-correction="true"
                    data-minimal-correction="paperShowCtrl.showMinimalCorrection()">
     </question-show>
-
+-->
```
Comment (Elorfin) [inline plugin/exo/Resources/modules/paper/Partials/show.html]: Remove commented code.

File: plugin/exo/Resources/modules/step/Controllers/StepShowCtrl.js [hunk 2]
```diff
@@ -32,6 +35,7 @@ function StepShowCtrl(UserPaperService, FeedbackService, QuestionService, StepSe
       this.solutionShown = true
     }
   }
+  this.getStepTotalScore()
```
Comment (Elorfin) [inline plugin/exo/Resources/modules/step/Controllers/StepShowCtrl.js]: Add a condition to calculate step score only if the feedback are visible or the correction is displayed.

File: plugin/exo/Resources/modules/step/Controllers/StepShowCtrl.js [hunk 3]
```diff
@@ -99,6 +125,13 @@ StepShowCtrl.prototype.showMinimalCorrection = function showMinimalCorrection()
 }
 
 /**
+ * On Feedback Hide
+ */
+StepShowCtrl.prototype.onFeedbackHide = function onFeedbackHide() {
+  this.stepScore = 0
+}
```
Comment (Elorfin) [inline plugin/exo/Resources/modules/step/Controllers/StepShowCtrl.js]: Not needed.

File: plugin/exo/Resources/modules/step/Partials/show.html [hunk 4]
```diff
@@ -3,11 +3,14 @@ <h4 class="step-title">
         {{ 'step'|trans:{}:'ujm_exo' }} <span class="badge">{{ stepShowCtrl.stepIndex*1 + 1 }}</span>
         <span data-ng-if="stepShowCtrl.step.meta.title">: {{ stepShowCtrl.step.meta.title }}</span>
 
-        <small data-ng-if="stepShowCtrl.step.meta.maxAttempts && stepShowCtrl.feedback.enabled" class="pull-right">
+
+        <small data-ng-if="!stepShowCtrl.solutionShown && stepShowCtrl.step.meta.maxAttempts && stepShowCtrl.feedback.enabled">
```
Comment (Elorfin) [inline plugin/exo/Resources/modules/step/Partials/show.html]: Change `stepShowCtrl.feedback.enabled` for `stepShowCtrl.feedback.visible`

File: plugin/exo/Resources/modules/step/Controllers/StepShowCtrl.js (hunk 2 above)
Comment (pitrackster) [inline plugin/exo/Resources/modules/step/Controllers/StepShowCtrl.js]: In fact this is the total score available for the step not the score of the answers 
If in a step there are 2 questions one for 2 points and one for 3 points this method give me 5 and step score is 0 / 5

File: plugin/exo/Resources/modules/paper/Partials/show.html (hunk 1 above)
Comment (pitrackster) [inline plugin/exo/Resources/modules/paper/Partials/show.html]: oups ...

File: plugin/exo/Resources/modules/step/Controllers/StepShowCtrl.js (hunk 2 above)
Comment (Elorfin) [inline plugin/exo/Resources/modules/step/Controllers/StepShowCtrl.js]: OK

**Comment by LaurentGruber** [comment]
LGTM

**Comment by Elorfin** [comment]
LGTM

**Pull request merged**